sdk/ai/ai-projects/samples-dev/agents/agents_bing_grounding_streaming.ts

```diff
@@ -0,0 +1,99 @@
+ *  BING_CONNECTION_NAME - the name of the connection with Bing search grounding
+ */
+
+import { AIProjectsClient, DoneEvent, ErrorEvent, MessageDeltaChunk, MessageDeltaTextContent, MessageStreamEvent, RunStreamEvent, ThreadRunOutput, fromConnectionId, connectionToolType } from "@azure/ai-projects"
+import { DefaultAzureCredential } from "@azure/identity";
+
+import * as dotenv from "dotenv";
+dotenv.config();
+
+const connectionString = process.env["AZURE_AI_PROJECTS_CONNECTION_STRING"] || "<endpoint>>;<subscription>;<resource group>;<project>";
+
+export async function main(): Promise<void> {
+  const client = AIProjectsClient.fromConnectionString(connectionString || "", new DefaultAzureCredential());
+  const bingConnection = await client.connections.getConnection(process.env["BING_CONNECTION_NAME"] || "<connection-name>");
+  const connectionId = bingConnection.id;
+
+  const bingTool = fromConnectionId(connectionToolType.BingGrounding, [connectionId]);
+
+  const agent  = await client.agents.createAgent(
+    "gpt-4-0125-preview", {
+      name: "my-agent", 
+      instructions: "You are a helpful agent",
+      tools: [bingTool]
+    }, {
+      headers: {"x-ms-enable-preview": "true"}
+    });
+  console.log(connectionId)
+  console.log(`Created agent, agent ID : ${agent.id}`);
+
+  const thread = await client.agents.createThread()
+  console.log(`Created thread, thread ID: ${thread.id}`);
+
+  const message = await client.agents.createMessage(thread.id, {role: "user", content: "How does wikipedia explain Euler's Identity?"});
+  console.log(`Created message, message ID: ${message.id}`);
+
+  const streamEventMessages = await client.agents.createRunStreaming(thread.id, agent.id);
+
+  for await (const eventMessage of streamEventMessages) {
+    switch (eventMessage.event) {
+      case RunStreamEvent.ThreadRunCreated:
+        console.log(`ThreadRun status: ${(eventMessage.data as ThreadRunOutput).status}`)
+        break;
+      case MessageStreamEvent.ThreadMessageDelta:
+        {
+          const messageDelta = eventMessage.data as MessageDeltaChunk;
+          messageDelta.delta.content.forEach((contentPart) => {
+            if (contentPart.type === "text") {
+              const textContent = contentPart as MessageDeltaTextContent
+              const textValue = textContent.text?.value || "No text"
+              console.log(`Text delta received:: ${textValue}`)
+            }
+          });
+        }
+        break;
+
+      case RunStreamEvent.ThreadRunCompleted:
+        console.log("Thread Run Completed");
+        break;
+      case ErrorEvent.Error:
+        console.log(`An error occurred. Data ${eventMessage.data}`);
+        break;
+      case DoneEvent.Done:
+        console.log("Stream completed.");
+        break;
+    }
+  }
+
+  client.agents.deleteAgent(agent.id)
+  console.log(`Deleted agent, agent ID: ${agent.id}`);
+
+  const messages = await client.agents.listMessages(thread.id)
```

Comment:
Is this still need for streaming ?

Reply:
The streaming logging isn't super readable since its a long response in this format:
<img width="147" alt="image" src="https://github.com/user-attachments/assets/ac2b93bf-5c90-4738-bb7c-9ddfce3d2f87">

Which is why I printed the messages in that form. Should I remove it?

Reply:
Its ok Keep it.